workflow/metrics/metrics.go

```diff
@@ -47,14 +48,15 @@ type Metrics struct {
 	defaultMetricDescs map[string]bool
 }
 
-var _ prometheus.Collector = Metrics{}
+var _ prometheus.Collector = &Metrics{}
 
-func New(metricsConfig, telemetryConfig ServerConfig) Metrics {
-	metrics := Metrics{
+func New(metricsConfig, telemetryConfig ServerConfig) *Metrics {
+	metrics := &Metrics{
 		metricsConfig:      metricsConfig,
 		telemetryConfig:    telemetryConfig,
 		workflowsProcessed: newCounter("workflows_processed_count", "Number of workflow updates processed", nil),
 		workflowsByPhase:   getWorkflowPhaseGauges(),
+		workflows:          make(map[string]bool),
```

Comment:
could you use the informer store?

Reply:
Great suggestion. I'll take a look

Reply:
Answer is no: the object is removed from the informer store before calling the handler function, making it impossible to distinguish if it's the first or second time the delete handler has been called.

Reply:
ok, I think we should have two changes, essentially this:

```
type workflowTracker struct {
  lock sync.Mutex
  workflows map[string]boo
}
```

That object can be responsible for tracking workflows - but before you do that - can we ask @jessesuen if he has a suggestion?